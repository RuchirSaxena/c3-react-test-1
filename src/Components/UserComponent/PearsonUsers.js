import React, { Component } from "react";
import User from "./Pearson";
import  removeDuplicateUsers  from "../../utility";
import { userDataApi } from  '../../Services/userDataApi' ;

const API_URL = 'https://reqres.in/api/users?page=1&per_page=10';

export class PearsonUsers extends Component {
  constructor(props) {
    super(props);

    this.state = {
      users: [
        {
          id: 4,
          first_name: "Eve",
          last_name: "Holt",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
        },
        {
          id: 5,
          first_name: "Charles",
          last_name: "Morris",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
        },
        {
          id: 6,
          first_name: "Tracey",
          last_name: "Ramos",
          avatar:
            "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
        }
      ],
      isloading: true,
      error: null
    };

    this.handleDeleteUser = this.handleDeleteUser.bind(this);
  }

  componentDidMount() {
    this.setState({ isLoading: true });
    userDataApi(API_URL)
      .then(response => {
        debugger;
        this.setState((prevState) => {
          return {
            users: removeDuplicateUsers([...prevState.users,...response.data]),
            isLoading: false
          }
        });
      })
      .catch(error => this.setState({ error, isLoading: false }));
  }
  handleDeleteUser(id) {
    this.setState((prevState) => {
      return {
        users: prevState.users.filter((user) => {
          return user.id !== id;
        })
      }
    });
  }
  render() {
    if (this.state.isLoading) {
      return <div className="loader"></div>;
    }
    const userData = this.state.users.map((user) => {
      return <li key={user.id}> <User  {...user} handleDeleteUser={this.handleDeleteUser} /> </li>
    });
    return (
      <div className="pearon-users">
        <h1>Pearson User Management</h1>
        <div className="box1">
          {this.state.users.length > 0 &&
            <ul>
              {userData}
            </ul>
          }
          {this.state.users.length === 0 &&
            <h2>No records found</h2>
          }

        </div>
      </div>
    );
  }
}
