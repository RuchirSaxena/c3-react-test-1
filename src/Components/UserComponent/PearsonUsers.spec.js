import { PearsonUsers } from './PearsonUsers';
import User from './Pearson';
import removeDuplicateUsers from "../../utility";

const pearsonUsersWrapper = shallow(<PearsonUsers />);

//Mocking user Api 
jest.mock("../../Services/userDataApi");


describe('PearsonUsers test cases', () => {

  it("fetches User data from user api and renders them on mount", (done) => {
    setTimeout(() => {
      pearsonUsersWrapper.update();
      const state = pearsonUsersWrapper.state();
      //checking state is updated after reciving user from api and delete the duplicate
      expect(state.users.length).toEqual(4);
      expect(pearsonUsersWrapper.find("User").length).toEqual(4);
      done();
    });
  });
  /**
   * Testing the state  after adding default user and users added through 
   * mocked Api
   */
  it('initializes the `state` with  list of `user`', () => {
    expect(pearsonUsersWrapper.state().users).toEqual([{
      id: 4,
      first_name: "Eve",
      last_name: "Holt",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
    },
    {
      id: 5,
      first_name: "Charles",
      last_name: "Morris",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
    },
    {
      id: 6,
      first_name: "Tracey",
      last_name: "Ramos",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
    },
    {
      "id": 1,
      "first_name": "George",
      "last_name": "Bluth",
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
    }
    ]);
  });

  //render a list of users from the state 
  it('render a list of users from the state correctly',()=>{
    expect(pearsonUsersWrapper).toMatchSnapshot();
  });

  /*
  test fuctionality to remove duplicated users from the state ,Also checkking the fuctionality of  removeDuplicateUsers() utility function to remove duplicates from array of object
  */
  it("To fuctionality to remove duplicated users from the state", () => {
    expect(removeDuplicateUsers(pearsonUsersWrapper.state().users)).toEqual([{
      id: 4,
      first_name: "Eve",
      last_name: "Holt",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
    },
    {
      id: 5,
      first_name: "Charles",
      last_name: "Morris",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
    },
    {
      id: 6,
      first_name: "Tracey",
      last_name: "Ramos",
      avatar:
        "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
    },
    {
      "id": 1,
      "first_name": "George",
      "last_name": "Bluth",
      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
    }
    ]);

  })

  //Test functionality to delete a user from the state.
  it("test functionality to delete a user from the state", () => {
    const id = 1;
    pearsonUsersWrapper.instance().handleDeleteUser(id);
    expect(pearsonUsersWrapper.state().users.length).toEqual(3);
  });

  //check handleDeleteUser fires on click
  it('check handleDeleteUser fires on click', () => {
    const spy = sinon.spy();
    const wrapper = mount(
      <User id={1} first_name={"George"} last_name={"Bluth"} avatar="128.jpg" handleDeleteUser={spy} />
    );

    wrapper
      .find("div").find("a")
      .first()
      .simulate("click");

    expect(spy.calledOnce).toBe(true);
  });



  //Match the snapshot for User Component
  it('matches the snapshot for User Component', () => {
    const tree = renderer.create(<User 
    id={1} 
    first_name={"George"}
    last_name={"Bluth"} avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
     handleDeleteUser={() => { }} />)
     .toJSON()

    expect(tree).toMatchSnapshot()
  });


})








