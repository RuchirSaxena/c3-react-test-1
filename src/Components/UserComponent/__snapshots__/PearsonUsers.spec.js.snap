// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PearsonUsers test cases matches the snapshot for User Component 1`] = `
<div>
  <img
    alt="person image"
    aria-hidden={true}
    src="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
  />
  <h3>
    George
     
    Bluth
  </h3>
  <a
    onClick={[Function]}
  >
    Delete
  </a>
</div>
`;

exports[`PearsonUsers test cases render a list of users from the state correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PearsonUsers />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Pearson User Management
</h1>,
        <div
          className="box1"
>
          <ul>
                    <li>
                               
                              <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                        first_name="Eve"
                                        handleDeleteUser={[Function]}
                                        id={4}
                                        last_name="Holt"
                              />
                               
                    </li>
                    <li>
                               
                              <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                        first_name="Charles"
                                        handleDeleteUser={[Function]}
                                        id={5}
                                        last_name="Morris"
                              />
                               
                    </li>
                    <li>
                               
                              <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                        first_name="Tracey"
                                        handleDeleteUser={[Function]}
                                        id={6}
                                        last_name="Ramos"
                              />
                               
                    </li>
                    <li>
                               
                              <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                        first_name="George"
                                        handleDeleteUser={[Function]}
                                        id={1}
                                        last_name="Bluth"
                              />
                               
                    </li>
          </ul>
</div>,
      ],
      "className": "pearon-users",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pearson User Management",
        },
        "ref": null,
        "rendered": "Pearson User Management",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ul>
              <li>
                             
                            <User
                                          avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                          first_name="Eve"
                                          handleDeleteUser={[Function]}
                                          id={4}
                                          last_name="Holt"
                            />
                             
              </li>
              <li>
                             
                            <User
                                          avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                          first_name="Charles"
                                          handleDeleteUser={[Function]}
                                          id={5}
                                          last_name="Morris"
                            />
                             
              </li>
              <li>
                             
                            <User
                                          avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                          first_name="Tracey"
                                          handleDeleteUser={[Function]}
                                          id={6}
                                          last_name="Ramos"
                            />
                             
              </li>
              <li>
                             
                            <User
                                          avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                          first_name="George"
                                          handleDeleteUser={[Function]}
                                          id={1}
                                          last_name="Bluth"
                            />
                             
              </li>
</ul>,
            false,
          ],
          "className": "box1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <li>
                   
                  <User
                                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                    first_name="Eve"
                                    handleDeleteUser={[Function]}
                                    id={4}
                                    last_name="Holt"
                  />
                   
</li>,
                <li>
                   
                  <User
                                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                    first_name="Charles"
                                    handleDeleteUser={[Function]}
                                    id={5}
                                    last_name="Morris"
                  />
                   
</li>,
                <li>
                   
                  <User
                                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                    first_name="Tracey"
                                    handleDeleteUser={[Function]}
                                    id={6}
                                    last_name="Ramos"
                  />
                   
</li>,
                <li>
                   
                  <User
                                    avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                    first_name="George"
                                    handleDeleteUser={[Function]}
                                    id={1}
                                    last_name="Bluth"
                  />
                   
</li>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": "4",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <User
                      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                      first_name="Eve"
                      handleDeleteUser={[Function]}
                      id={4}
                      last_name="Holt"
/>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                      "first_name": "Eve",
                      "handleDeleteUser": [Function],
                      "id": 4,
                      "last_name": "Holt",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "5",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <User
                      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                      first_name="Charles"
                      handleDeleteUser={[Function]}
                      id={5}
                      last_name="Morris"
/>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                      "first_name": "Charles",
                      "handleDeleteUser": [Function],
                      "id": 5,
                      "last_name": "Morris",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "6",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <User
                      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                      first_name="Tracey"
                      handleDeleteUser={[Function]}
                      id={6}
                      last_name="Ramos"
/>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                      "first_name": "Tracey",
                      "handleDeleteUser": [Function],
                      "id": 6,
                      "last_name": "Ramos",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                ],
                "type": "li",
              },
              Object {
                "instance": null,
                "key": "1",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    <User
                      avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                      first_name="George"
                      handleDeleteUser={[Function]}
                      id={1}
                      last_name="Bluth"
/>,
                    " ",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
                      "first_name": "George",
                      "handleDeleteUser": [Function],
                      "id": 1,
                      "last_name": "Bluth",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  " ",
                ],
                "type": "li",
              },
            ],
            "type": "ul",
          },
          false,
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Pearson User Management
</h1>,
          <div
            className="box1"
>
            <ul>
                        <li>
                                     
                                    <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                                first_name="Eve"
                                                handleDeleteUser={[Function]}
                                                id={4}
                                                last_name="Holt"
                                    />
                                     
                        </li>
                        <li>
                                     
                                    <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                                first_name="Charles"
                                                handleDeleteUser={[Function]}
                                                id={5}
                                                last_name="Morris"
                                    />
                                     
                        </li>
                        <li>
                                     
                                    <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                                first_name="Tracey"
                                                handleDeleteUser={[Function]}
                                                id={6}
                                                last_name="Ramos"
                                    />
                                     
                        </li>
                        <li>
                                     
                                    <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                                first_name="George"
                                                handleDeleteUser={[Function]}
                                                id={1}
                                                last_name="Bluth"
                                    />
                                     
                        </li>
            </ul>
</div>,
        ],
        "className": "pearon-users",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pearson User Management",
          },
          "ref": null,
          "rendered": "Pearson User Management",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ul>
                <li>
                                 
                                <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                                first_name="Eve"
                                                handleDeleteUser={[Function]}
                                                id={4}
                                                last_name="Holt"
                                />
                                 
                </li>
                <li>
                                 
                                <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                                first_name="Charles"
                                                handleDeleteUser={[Function]}
                                                id={5}
                                                last_name="Morris"
                                />
                                 
                </li>
                <li>
                                 
                                <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                                first_name="Tracey"
                                                handleDeleteUser={[Function]}
                                                id={6}
                                                last_name="Ramos"
                                />
                                 
                </li>
                <li>
                                 
                                <User
                                                avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                                first_name="George"
                                                handleDeleteUser={[Function]}
                                                id={1}
                                                last_name="Bluth"
                                />
                                 
                </li>
</ul>,
              false,
            ],
            "className": "box1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <li>
                     
                    <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                                        first_name="Eve"
                                        handleDeleteUser={[Function]}
                                        id={4}
                                        last_name="Holt"
                    />
                     
</li>,
                  <li>
                     
                    <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                                        first_name="Charles"
                                        handleDeleteUser={[Function]}
                                        id={5}
                                        last_name="Morris"
                    />
                     
</li>,
                  <li>
                     
                    <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                                        first_name="Tracey"
                                        handleDeleteUser={[Function]}
                                        id={6}
                                        last_name="Ramos"
                    />
                     
</li>,
                  <li>
                     
                    <User
                                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                                        first_name="George"
                                        handleDeleteUser={[Function]}
                                        id={1}
                                        last_name="Bluth"
                    />
                     
</li>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": "4",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <User
                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg"
                        first_name="Eve"
                        handleDeleteUser={[Function]}
                        id={4}
                        last_name="Holt"
/>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/marcoramires/128.jpg",
                        "first_name": "Eve",
                        "handleDeleteUser": [Function],
                        "id": 4,
                        "last_name": "Holt",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "5",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <User
                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg"
                        first_name="Charles"
                        handleDeleteUser={[Function]}
                        id={5}
                        last_name="Morris"
/>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/stephenmoon/128.jpg",
                        "first_name": "Charles",
                        "handleDeleteUser": [Function],
                        "id": 5,
                        "last_name": "Morris",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "6",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <User
                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg"
                        first_name="Tracey"
                        handleDeleteUser={[Function]}
                        id={6}
                        last_name="Ramos"
/>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/bigmancho/128.jpg",
                        "first_name": "Tracey",
                        "handleDeleteUser": [Function],
                        "id": 6,
                        "last_name": "Ramos",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                  ],
                  "type": "li",
                },
                Object {
                  "instance": null,
                  "key": "1",
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      " ",
                      <User
                        avatar="https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"
                        first_name="George"
                        handleDeleteUser={[Function]}
                        id={1}
                        last_name="Bluth"
/>,
                      " ",
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "avatar": "https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg",
                        "first_name": "George",
                        "handleDeleteUser": [Function],
                        "id": 1,
                        "last_name": "Bluth",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": [Function],
                    },
                    " ",
                  ],
                  "type": "li",
                },
              ],
              "type": "ul",
            },
            false,
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
